### PR TITLE
[24.1] Show items with same `hid` but different `id` in history

### PR DESCRIPTION
**Pull request by ahmedhamidawan**
This adds a `sub_items` array to history items when they are fetched in the `historyItemsStore`, so that if any history item has other related items with the same hid, they are pushed to this array, and can be shown in the history.

This is based on the current filter and what the `contents` call is returning. If the API is not returning compressed versions of this dataset (for the current filter), then we don't show the "Show converted items" icon until those items are fetched too.

Is this okay, or do we want to go with the expand dataset and see converted versions of it below?

### The expand dataset to see converted items approach:
Problem with the expand-to-fetch approach is that we would have to make another API call to find any other datasets with same hid, different id for each dataset on expansion (other than the `fetchDataset` call right now), and it wouldn't show the "Show converted" icon by default until the user expands the dataset.

### The current implementation (based on filter):
The advantage with doing it the way it is now is that the `HistoryCounter` showing 13 items while 12 are visible is partially solved now because we see the little number on the converted items icon on the dataset (in this e.g.:`1`), proving to the user that the counter is correct - until the user clears the filter and the converted-counter is still there (because we don't reset the `sub_items` array for the new filter).

https://github.com/user-attachments/assets/8c20f497-bc61-438e-bdb5-cabc89c7969f

Fixes https://github.com/galaxyproject/galaxy/issues/17220

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by ahmedhamidawan** [comment]
### UI Meeting Notes:
- Use `create_time` to confirm which item is the original dataset, and which is converted
- "Show converted items" instead of "Show associated items"
- Change plus icon to converted (`faExchangeAlt` ... we do use this for history switching,,,)
- Only show when dataset is expanded, fetch items with same hid from backend on that call

**Comment by ahmedhamidawan** [comment]
@mvdbeek 
Unfortunately, if your history is imported, the converted (or compressed) items will have the same `create_time` as the main item. So, in that case, is it fine to by default, show whichever one happens to be first in the `contents` API?
If not, and we must show the original one by default, we would still need a way to know explicitly which items are compressed.

**Comment by mvdbeek** [comment]
That's an old bug that should be fixed now.

**Comment by ahmedhamidawan** [comment]
> That's an old bug that should be fixed now.

It's still there on main now (i imported the history you shared):

https://github.com/user-attachments/assets/f5702571-f038-4da5-9b7e-302a05b7d68d

**Comment by mvdbeek** [comment]
Ah, yes, the bug fixed was that the original history update time was updated. Can you work with the create time anyway ? If you run a new implicit conversion you won't have the same create time.

**Comment by ahmedhamidawan** [comment]
But once the history is imported, the `create_time` for all the items is the same as the time it was imported: http://localhost:8081/u/ahmedhamidawan/h/marius-17220-test-imported

**Comment by mvdbeek** [comment]
I'm saying you can ignore this for now ?

File: client/src/components/History/Content/ContentItem.vue [hunk 1]
```diff
@@ -287,6 +298,7 @@ function onUndelete() {
 }
 
 function onDragStart(evt: DragEvent) {
+    console.log(evt);
```
Comment (mvdbeek) [inline client/src/components/History/Content/ContentItem.vue]:
```suggestion
```

**Comment by mvdbeek** [comment]
@ahmedhamidawan can you rebase this against 24.1 ? This has been broken since forever and it's not a small change, I would prefer to just target the last release here.

**Pull request merged**